osu.Game.Rulesets.Osu.Tests/TestCaseSliderInput.cs

```diff
@@ -0,0 +1,240 @@
+            AddAssert("Tracking lost", assertMehJudge);
+        }
+
+        /// <summary>
+        ///     Hitting a slider head, pressing a new key after the initial hit, then letting go of the original key used to hit
+        ///     the slider should reslt in continued tracking.
+        ///     At 250ms intervals:
+        ///     Frame 1: Left Click
+        ///     Frame 2: Left & Right Click
+        ///     Frame 3: Right Click
+        ///     A passing test case will have the cursor continue to track after frame 3.
+        /// </summary>
+        [Test]
+        public void TestLeftBeforeSliderThenRightThenLettingGoOfLeft()
+        {
+            AddStep("Left to both to right test", () =>
+            {
+                var actions = new List<List<OsuAction>>
+                {
+                    new List<OsuAction> { OsuAction.LeftButton },
+                    new List<OsuAction> { OsuAction.LeftButton, OsuAction.RightButton },
+                    new List<OsuAction> { OsuAction.RightButton }
+                };
+
+                performStaticInputTest(actions);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 2");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        /// <summary>
+        ///     Hitting a slider head, pressing a new key after the initial hit, then letting go of the new key should result
+        ///     in continue tracking,
+        ///     At 250ms intervals:
+        ///     Frame 1: Left Click
+        ///     Frame 2: Left & Right Click
+        ///     Frame 3: Right Click
+        ///     A passing test case will have the cursor continue to track after frame 3.
+        /// </summary>
+        [Test]
+        public void TestTrackingRetentionLeftRightLeft()
+        {
+            AddStep("Tracking retention test", () =>
+            {
+                var actions = new List<List<OsuAction>>
+                {
+                    new List<OsuAction> { OsuAction.LeftButton },
+                    new List<OsuAction> { OsuAction.LeftButton, OsuAction.RightButton },
+                    new List<OsuAction> { OsuAction.LeftButton }
+                };
+
+                performStaticInputTest(actions);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 3");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        /// <summary>
+        ///     Pressing a key before a slider, pressing the other key on the slider head, then releasing the former pressed key
+        ///     should result in continued tracking.
+        ///     At 250ms intervals:
+        ///     Frame 1: Left Click
+        ///     Frame 2: Left & Right Click
+        ///     Frame 3: Right Click
+        ///     A passing test case will have the cursor continue to track after frame 3.
+        /// </summary>
+        [Test]
+        public void TestTrackingLeftBeforeSliderToRight()
```

Comment:
this test seems to do exactly the same as the one below it?

Reply:
As discussed, will make this method more readable by pairing timestamps with inputs